### ISSUE TITLE
Create TAR indexer

### ISSUE THREAD

**Issue by ovidiul** [ovidiul/XCloner-Wordpress#4]
Create a TAR index file that can be appended at the end of the tar archive which should contain the file path and archive position for quicker file retrieval.

Create a method to retrieve a file starting from the end of the files archive.

**Comment by vdrover** [comment]
@ovidiul is this still relevant?

**Comment by ovidiul** [comment]
Yes, right now, when we want for instance to extract the database file which is usually at the end, we have to parse the full backup archive until we reach that file, not knowing what's next(this goes specifically for doing a DB only restore). 

The impediment here for the user would be the long wait time to restore a single file.

Having a list at the end of the archive, would probably save some time in restoring single files from a larger archive as we could quickly fetch the required file position in the archive and then just fetch it starting from that position.

**Comment by vdrover** [comment]
Before we take this any further, what is the current workflow to restore a single file (assuming this exists).
